pages/api/dpu/trigger.ts

```diff
@@ -2,22 +2,35 @@ import { NextApiRequest, NextApiResponse } from "next";
 import { saveTopicNameInUsersTable, createTopicName } from '../../../utils/db/relevance';
 import { createTopicNameInGcloud } from '../../../utils/pubsub/pubsubClient';
 import { CloudBuildStatus, triggerCloudProjectBuildUsingGcloudApi, pollBuildStatus, triggerCloudPatBuildUsingGcloudApi } from './../../../utils/trigger';
-import { DbUser, getUserById } from '../../../utils/db/users';
+import { DbUser, getUserById, getUserIdByTopicName } from '../../../utils/db/users';
+import rudderStackEvents from '../events';
 
 const triggerHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 	console.info("[triggerHandler] pub sub setup info in db...");
 	const jsonBody = req.body;
 	let topicName: string;
+
+	const event_properties = {
+		repo_owner: jsonBody.repo_owner || "",
+		repo_provider: jsonBody.selectedProvider || "",
+		installation_type: jsonBody.selectedInstallationType || "",
+		hosting: jsonBody.selectedHosting || "",
+	};
+
 	if (!jsonBody.selectedProvider || !jsonBody.selectedInstallationType || !jsonBody.selectedHosting || !jsonBody.userId) {
 		console.error("[triggerHandler] Invalid request body");
 		res.status(400).json({ "error": "Invalid request body" });
+		const eventProperties = { ...event_properties, response_status: 400 };
+		rudderStackEvents.track("absent", "", 'dpu-trigger', { type: 'HTTP-400', eventStatusFlag: 0, eventProperties });
```

Comment:
```suggestion
		rudderStackEvents.track(jsonBody?.userId ?? "absent", "", 'dpu-trigger', { type: 'HTTP-400', eventStatusFlag: 0, eventProperties });
```
It is possible that `userId` exists.